frontend/src/components/guards/RequireAuth.vue

```diff
@@ -0,0 +1,38 @@
+<script setup lang="ts">
+import { storeToRefs } from 'pinia';
+import { onBeforeMount, ref } from 'vue';
+import { useRouter } from 'vue-router';
+
+import { useAuthStore } from '@/store';
+import { RouteNames } from '@/utils/constants';
+
+import type { Ref } from 'vue';
+
+// Store
+const { getIsAuthenticated } = storeToRefs(useAuthStore());
+
+// State
+const ready: Ref<boolean> = ref(false);
+
+// Actions
+onBeforeMount( async () => {
+  const router = useRouter();
```

Comment:
Should this be outside the onBeforeMount callback an in the general setup function instead?